reddit.js

```diff
@@ -1,5 +1,63 @@
 var reddit = require("./data.json")
+var datas = reddit["data"];
+var dataChild = datas["children"];
+//console.log(dataChild);
+//console.log(dataChild[0].data);
 
 // Write your solutions below
 
-console.log(reddit["data"]);
\ No newline at end of file
+//for each function
+var forE = function(){
+	dataChild.forEach (function(item){
+		console.log(item.data.title + "is the title");
+	});
+}
+
+var perma = function() {
+	var links = [];
+	//this pushes all permalinks into an array
+	dataChild.forEach ( function(perms){
+		links.push(perms.data.permalink);
+	});
+	// this maps them with http://www.reddit.com
+	var link2 = links.map(function(sub){
+		return "http://www.reddit.com"+sub;
+	})
+	console.log(link2);
+}
+
+var filt = function(){
+	//returns array of objects which are not blank
+	var self = dataChild.filter(function(selfy){
+		return selfy.data.selftext != "";
+	});
+	//prints titles in array
+	for (posts in self){
+		console.log("posts with selftext include:")
+		console.log(self[posts].data.title);
+	}
+	
+}
+
+var redux = function(){
+	var scores =[];
```

Comment:
Making a scoreArray wasn't a bad idea, but we can actually just call reduce on children here.
```
var sum = children.reduce(function(runningTotal, child, index) {
    return runningTotal + child.data.score
}, 0);
console.log(sum);
```